R/learner_mgcv_regr_gam.R

```diff
@@ -0,0 +1,195 @@
+#'
+#' @export
+LearnerRegrGam = R6Class("LearnerRegrGam",
+  inherit = LearnerRegr,
+
+  public = list(
+    #' @description
+    #' Creates a new instance of this [R6][R6::R6Class] class.
+    initialize = function() {
+      ps = ParamSet$new(list(
+        ParamFct$new(
+          "family",
+          default = "gaussian",
+          levels = c("gaussian", "poisson"),
+          tags = "train"
+        ),
+        ParamUty$new("formula", default = NULL, tags = "train"),
+        ParamUty$new("offset", default = NULL, tags = "train"),
+        ParamFct$new(
+          "method",
+          levels = c("GCV.Cp", "GACV.Cp", "REML", "P-REML", "ML", "P-ML"),
+          default = "GCV.Cp",
+          tags = "train"
+        ),
+        ParamUty$new("optimizer", default = c("outer", "newton"), tags = "train"),
+        ParamDbl$new("scale", default = 0, tags = "train"),
+        ParamLgl$new("select", default = FALSE, tags = "train"),
+        ParamUty$new("knots", default = NULL, tags = "train"),
+        ParamUty$new("sp", default = NULL, tags = "train"),
+        ParamUty$new("min.sp", default = NULL, tags = "train"),
+        ParamUty$new("H", default = NULL, tags = "train"),
+        ParamDbl$new("gamma", default = 1, lower = 1, tags = "train"),
+        ParamUty$new("paraPen", default = NULL, tags = "train"),
+        ParamUty$new("G", default = NULL, tags = "train"),
+        ParamUty$new("in.out", default = NULL, tags = "train"),
+        ParamLgl$new("drop.unused.levels", default = TRUE, tags = "train"),
+        ParamLgl$new("drop.intercept", default = FALSE, tags = "train"),
+        ParamInt$new("nthreads", default = 1L, lower = 1L, tags = c("train", "control")),
+        ParamDbl$new("irls.reg", default = 0.0, lower = 0, tags = c("train", "control")),
+        ParamDbl$new("epsilon", default = 1e-07, lower = 0, tags = c("train", "control")),
+        ParamInt$new("maxit", default = 200L, tags = c("train", "control")),
+        ParamLgl$new("trace", default = FALSE, tags = c("train", "control")),
+        ParamDbl$new("mgcv.tol", default = 1e-07, lower = 0, tags = c("train", "control")),
+        ParamInt$new("mgcv.half", default = 15L, lower = 0L, tags = c("train", "control")),
+        ParamDbl$new(
+          "rank.tol",
+          default = .Machine$double.eps^0.5,
+          lower = 0,
+          tags = c("train", "control")
+        ),
+        ParamUty$new("nlm", default = list(), tags = c("train", "control")),
+        ParamUty$new("optim", default = list(), tags = c("train", "control")),
+        ParamUty$new("newton", default = list(), tags = c("train", "control")),
+        ParamInt$new("outerPIsteps", default = 0L, lower = 0L, tags = c("train", "control")),
+        ParamLgl$new("idLinksBases", default = TRUE, tags = c("train", "control")),
+        ParamLgl$new("scalePenalty", default = TRUE, tags = c("train", "control")),
+        ParamInt$new("efs.lspmax", default = 15L, lower = 0L, tags = c("train", "control")),
```

Comment:
I have no idea why codefactor is running here it's clearly not a constant! Let's just nolint it:

```suggestion
        ParamInt$new("efs.lspmax", default = 15L, lower = 0L, tags = c("train", "control")), # nolint
```